test/functional/wallet-hd.py

```diff
@@ -91,6 +91,21 @@ def run_test (self):
         self.start_node(1, extra_args=self.extra_args[1] + ['-rescan'])
         assert_equal(self.nodes[1].getbalance(), num_hd_adds + 1)
 
+        # Try a RPC based rescan
+        self.stop_node(1)
+        shutil.rmtree(tmpdir + "/node1/regtest/blocks")
```

Comment:
Really, this should use `os.path.join` to construct the path, but I see this pattern is used in other tests, so I don't think you need to change it here.

Even better would be to move these functions to be methods in the `TestNode` class - `remove_block_files()`, `remove_chainstate_files()`, `backup_wallet_file()`, `restore_wallet_file()`, etc.